components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -444,6 +448,24 @@ public static API importApi(String extractedFolderPath, APIDTO importedApiDTO, B
         return operationPoliciesMap;
     }
 
+    public static List<OperationPolicy> extractValidateAndDropAPIPoliciesFromAPI(APIDTO importedApiDTO,
+             String extractedFolderPath, String apiUUID, String tenantDomain,
+             String apiType, APIProvider provider) throws APIManagementException {
+        List<OperationPolicy> apiPoliciesList = new ArrayList<>();
+        if (importedApiDTO.getApiPolicies() != null) {
+            apiPoliciesList = OperationPolicyMappingUtil
+                    .fromDTOToAPIOperationPoliciesList(importedApiDTO.getApiPolicies());
+            Map<String, OperationPolicySpecification> visitedPoliciesMap = new HashMap<>();
+            for (OperationPolicy policy : apiPoliciesList) {
+                validateAppliedPolicy(policy, visitedPoliciesMap, extractedFolderPath, apiUUID, provider,
+                        tenantDomain, apiType);
+            }
+
+        }
+        importedApiDTO.setApiPolicies(null);
```

Comment:
Is this a typo ? What is the need to set apiPolicies of apiDTO to null ?

Reply:
We extract, validate and drop the policies from the API object as to record policy mapping, we need api uuid. API will be created without policies and after that api will be updated.